### PR TITLE
Do not make the service name in the header a link if no `serviceUrl` is provided

### LINKED ISSUE #1826: Prevent creating `<a>` without `href` in header service name
## What
As discussed in https://github.com/alphagov/govuk-frontend/pull/1825, `serviceName` has an implicit dependency on `serviceUrl`. If `serviceName` is set but `serviceUrl` isn't, the macro renders an `<a>` tag without `href`. 

If `serviceName` is set, we could:
- Check if `serviceUrl` is set and if it's not, render a `<span>` in place of `<a>` OR have a default value `/` for `href` 
- Improve the macro docs around this to make the dependency explicit.

## Why
To help our users to avoid outputting invalid markup / miss including the href. 

## Who needs to know about this
Devs, Mark?

## Done when
- [ ] It's not possible to output an `<a>` tag without `href` for service name using the macro 
- [ ] We've improved the macro documentation to make the dependency clearer

### PR DESCRIPTION
**Pull request by 36degrees**
The header macro currently assumes that if there is a serviceName then there will always be a serviceUrl. If no serviceUrl is provided then the service name is wrapped with a link with an empty `href` attribute.

There’s not always a logical place to link the service name in the header to, so it makes sense to allow for service names that don’t link anywhere.

Update the macro so that when serviceUrl is omitted from the options a span is used to wrap the service name rather than a link.

As the current class used on the service name is a modifier for links (`govuk-header__link--service-name`) but will not be used on a link in this context, rename it to `govuk-header__service-name` to make it more generic.

Keep the old class name around as an alias to prevent this being a breaking change, but mark it as deprecated.

Fixes #1826

### CODE REVIEW

File: src/govuk/components/header/_index.scss [hunk 1]
```diff
@@ -142,6 +142,9 @@
     }
   }
 
+  // The govuk-header__link--service-name class is deprecated - use
+  // govuk-header__service-name instead.
+  .govuk-header__service-name,
```
Comment (36degrees) [inline src/govuk/components/header/_index.scss:147]: Are we happy with this approach? If so, will need to create an issue to remove the `govuk-header__link--service-name` class and attach it to the milestone for v5.0.

The alternative I think is to either:

- live with having a link modifier class on something that isn't a link (which feels icky)
- keep both classes around forever

**Review by owenatgov — APPROVED** [review]
Looking fab. I'm happy with the approach and am in favour of us removing `govuk-header__link--service-name`. It makes sense to keep our styles clean at least in the long term.

**Comment by owenatgov** [comment]
Afterthought: Does this need a changelog entry?

File: src/govuk/components/header/_index.scss (hunk 1 above)
Comment (36degrees) [reply src/govuk/components/header/_index.scss:147]: I've created an issue to remove the deprecated class here: https://github.com/alphagov/govuk-frontend/issues/2636

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,22 @@
 
 ## Unreleased
 
+### Recommended changes
+
+#### Replace deprecated `govuk-header__link--service-name` class in the header
+
+If you're not using the Nunjucks macros, replace any instances of the class `govuk-header__link--service-name` with `govuk-header__service-name` in the header.
```
Comment (EoinShaughnessy) [inline CHANGELOG.md:9]:
```suggestion
If you're not using the Nunjucks macros, replace any instances in the header of the class `govuk-header__link--service-name` with `govuk-header__service-name`.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,22 @@
...
+
+The `govuk-header__link--service-name` class has been deprecated and will be removed in a future major release.
```
Comment (EoinShaughnessy) [inline CHANGELOG.md:11]:
```suggestion
We've deprecated the `govuk-header__link--service-name` class, and will remove it in a future major release.
```

**Review by EoinShaughnessy — COMMENTED** [review]
Content looks good! Made 2 small suggestions, but not feeling strongly about either so feel free to accept/ignore.

File: CHANGELOG.md (hunk 3 above)
Comment (EoinShaughnessy) [reply CHANGELOG.md]: 🤣

**Pull request merged**